File: GFG/JavaScript/Easy/search-in-a-matrix.js
// https://www.geeksforgeeks.org/problems/search-in-a-matrix17201720/1

class Solution {
  matSearch(matrix, N, M, X) {
    let height = 0;
    let width = M - 1;

    while (height <= N - 1 && width >= 0) {
      if (matrix[height][width] == X) return 1;
      if (matrix[height][width] > X) width--;
      if (matrix[height][width] < X) height++;
    }
    return 0;
  }
}

File: GFG/Python/Easy/search-in-a-matrix.py
class Solution:
    def mat_search(self, mat, N, M, X):
        # Complete this function
        height = 0
        width = M-1

        while height <= N-1 and width >= 0:
            if mat[height][width] == X:
                return 1
            if mat[height][width] > X:
                width -= 1
            else:
                height += 1

        return 0
